HashTable.java

```diff
@@ -0,0 +1,192 @@
+import java.util.*;
+import java.lang.reflect.Array;
+
+class HashTable <K, V> {
+
+    // The array cannot be smaller than MIN_CAPACITY at any time
+    private static final int MIN_CAPACITY = 32;
+    private int approxSize = Integer.highestOneBit(MIN_CAPACITY) << 1;
+    private int tableMask = approxSize - 1;
+
+    private HashEntry[] table;
+
+    // The current number of elements.
+    private int size;
+
+    // The current capacity of the array.
+    private int capacity;
+
+    // initialLoad determines how full the array should be made when resizing
+    private double initialLoad;
+    private double maxLoad;
+    private double minLoad;
+
+    public HashTable(){
+        size = 0;
+        capacity = MIN_CAPACITY;
+        maxLoad = 0.75;
+        minLoad = 0.25;
+        this.initialLoad = 0.5;
+
+        //HashEntry[] table = new HashEntry[capacity];
+        table = (HashEntry[]) Array.newInstance(HashEntry.class, capacity);
+    }
+
+    /**
+    * Returns the value associated with the specified key.
+    * @param key the key
```

Comment:
Add a blank line with only the `*` between the first sentence and the `@param` entries.